components/draw/Toolbox.svelte

```diff
@@ -0,0 +1,96 @@
+<script>
+  import SelectMode from "./SelectMode.svelte";
+  import EditMode from "./EditMode.svelte";
+  import RouteMode from "./RouteMode.svelte";
+  import PointOrPolygonMode from "./PointOrPolygonMode.svelte";
+
+  export let routeUrl;
+  // Plumbed up from RouteMode, so we can pass it down to EditMode
+  let routeSnapper;
+  let snapTool;
+
+  let mode = "select";
+
+  function selectMode() {
+    mode = "select";
+  }
+  function editMode() {
+    mode = "edit";
+  }
+  // TODO For these and route, disable sidebar interactions
+  function newPointMode() {
+    mode = "point";
+  }
+  function newPolygonMode() {
+    mode = "polygon";
+  }
+</script>
+
+<div class="toolbox">
+  <div>
+    <button type="button" on:click={selectMode} disabled={mode == "select"}
+      >Select</button
+    >
+    <SelectMode {mode} />
+  </div>
+  <div>
+    <button type="button" on:click={editMode} disabled={mode == "edit"}
+      >Edit</button
+    >
+    {#if routeSnapper && snapTool}
+      <EditMode bind:mode {routeSnapper} {snapTool} />
+    {/if}
+  </div>
+  <div>
+    <button type="button" on:click={newPointMode} disabled={mode == "point"}
+      >New point</button
+    >
+    <PointOrPolygonMode bind:mode type="point" />
+  </div>
+  <div>
+    <button type="button" on:click={newPolygonMode} disabled={mode == "polygon"}
+      >New polygon</button
+    >
+    <PointOrPolygonMode bind:mode type="polygon" />
+  </div>
+  <div>
+    <RouteMode bind:mode url={routeUrl} bind:routeSnapper bind:snapTool />
+  </div>
+</div>
+
+<style>
+  .toolbox {
+    position: absolute;
+    top: 10px;
+    right: 10px;
+    padding: 10px;
+    background-color: white;
+    border: solid 2px black;
+    font-size: 1.5em;
+  }
+
+  button {
+    background-color: #4caf50;
+    border: none;
+    color: white;
+    padding: 15px 32px;
+    text-align: center;
+    text-decoration: none;
+    display: inline-block;
+    font-size: 16px;
+    margin: 4px 2px;
+    cursor: pointer;
```

Comment:
Re my comment about text wrapping I tried adding 
`overflow-wrap: break-word;` here but this doesn't work. 
I think because that requires an absolute width to cause an overflow which isn't what we see here.

Reply:
I think it'd be the `div` above, not the `button`. I'll play around tomorrow. The styling in the toolbox is pretty placeholder right now. I also want to overhaul the route tool, so it doesn't look so wildly different!